src/store/index-level.ts

```diff
@@ -120,27 +123,77 @@ export class IndexLevel {
     }
 
     // delete the reverse lookup
-    const partitionOperation = await this.createOperationForIndexesLookupPartition(tenant, { type: 'del', key: messageCid });
-    indexOps.push(partitionOperation);
+    const partitionOperationPromise = this.createOperationForIndexesLookupPartition(tenant, { type: 'del', key: messageCid });
+    indexOpPromises.push(partitionOperationPromise);
 
-    // delete the keys for each sortIndex
+    // delete the keys for each index
     for (const indexName in indexes) {
-      const sortValue = indexes[indexName];
-      const partitionOperation = await this.createOperationForIndexPartition(
-        tenant,
-        indexName,
-        {
-          type : 'del',
-          key  : IndexLevel.keySegmentJoin(IndexLevel.encodeValue(sortValue), messageCid)
+      const indexValue = indexes[indexName];
+      if (Array.isArray(indexValue)) {
+        for (const indexValueItem of indexValue) {
+          const partitionOperationPromise = this.generateDeleteIndexedItemOperation(tenant, messageCid, indexName, indexValueItem);
+          indexOpPromises.push(partitionOperationPromise);
         }
-      );
-      indexOps.push(partitionOperation);
+        continue;
+      }
+      const partitionOperationPromise = this.generateDeleteIndexedItemOperation(tenant, messageCid, indexName, indexValue);
+      indexOpPromises.push(partitionOperationPromise);
     }
 
+    const indexOps = await Promise.all(indexOpPromises);
     const tenantPartition = await this.db.partition(tenant);
     await tenantPartition.batch(indexOps, options);
   }
 
+  /**
+   * Creates an IndexLevel `put` operation for indexing an item, creating a partition by `tenant` and by `indexName`
+   *
+   * The key is indexValue followed by the messageCid as a tie-breaker.
+   * for example if the property is messageTimestamp the key would look like:
+   * '"2023-05-25T18:23:29.425008Z"\u0000bafyreigs3em7lrclhntzhgvkrf75j2muk6e7ypq3lrw3ffgcpyazyw6pry'
+   *
+   * The value is the JSON string of the `item`
```

Comment:
Consider moving this context info within the implementation (line 165), same for `generateDeleteIndexedItemOperation()`.